flink-core/src/main/java/org/apache/flink/api/java/typeutils/PojoTypeInfo.java

```diff
@@ -70,10 +75,21 @@
 	private static final Pattern PATTERN_NESTED_FIELDS = Pattern.compile(REGEX_NESTED_FIELDS);
 	private static final Pattern PATTERN_NESTED_FIELDS_WILDCARD = Pattern.compile(REGEX_NESTED_FIELDS_WILDCARD);
 
+	private static final Map<Class<?>, Class<? extends TypeSerializer>> customSerializers = new HashMap<>();
+	private static final Map<Class<?>, Class<? extends TypeComparator>> customComparators = new HashMap<>();
+
 	private final PojoField[] fields;
 	
 	private final int totalFields;
 
+	public static <C, S extends TypeSerializer<C>> void registerCustomSerializer(Class<C> c, Class<S> s) {
```

Comment:
Still needs to be documented (and `registerCustomComparator`, but I wanted to note that it would be good to document the order used in `createSerializer` ... Kyro, Avro, custom, code-generated.